tests/components/brottsplatskartan/test_config_flow.py

```diff
@@ -0,0 +1,224 @@
+        "app_id": "ha-1234567890",
+    }
+
+
+async def test_import_flow_success(hass: HomeAssistant) -> None:
+    """Test a successful import of yaml."""
+
+    with patch(
+        "homeassistant.components.brottsplatskartan.sensor.BrottsplatsKartan",
+    ), patch(
+        "homeassistant.components.brottsplatskartan.config_flow.uuid.getnode",
+        return_value="1234567890",
+    ), patch(
+        "homeassistant.components.brottsplatskartan.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={},
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "Brottsplatskartan HOME"
+    assert result2["data"] == {
+        "latitude": None,
+        "longitude": None,
+        "area": "N/A",
+        "app_id": "ha-1234567890",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_flow_location_success(hass: HomeAssistant) -> None:
+    """Test a successful import of yaml with location."""
+
+    with patch(
+        "homeassistant.components.brottsplatskartan.sensor.BrottsplatsKartan",
+    ), patch(
+        "homeassistant.components.brottsplatskartan.config_flow.uuid.getnode",
+        return_value="1234567890",
+    ), patch(
+        "homeassistant.components.brottsplatskartan.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                CONF_LATITUDE: 59.32,
+                CONF_LONGITUDE: 18.06,
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == RESULT_TYPE_CREATE_ENTRY
+    assert result2["title"] == "Brottsplatskartan 59.32, 18.06"
+    assert result2["data"] == {
+        "latitude": 59.32,
+        "longitude": 18.06,
+        "area": "N/A",
+        "app_id": "ha-1234567890",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_flow_already_exist(hass: HomeAssistant) -> None:
+    """Test import of yaml already exist."""
+
+    MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            "latitude": None,
+            "longitude": None,
+            "area": "N/A",
+            "app_id": "ha-1234567890",
+        },
+        unique_id="bpk-home",
+    ).add_to_hass(hass)
+
+    with patch(
+        "homeassistant.components.brottsplatskartan.sensor.BrottsplatsKartan",
+    ), patch(
+        "homeassistant.components.brottsplatskartan.config_flow.uuid.getnode",
+        return_value="1234567890",
+    ), patch(
+        "homeassistant.components.trafikverket_train.async_setup_entry",
+        return_value=True,
+    ):
```

Comment:
Not needed if `pytestmark = pytest.mark.usefixtures("mock_setup_entry")` is used.